jsonrpc/eth_endpoint.go

```diff
@@ -155,6 +155,41 @@ func (e *Eth) GetBlockByHash(hash types.Hash, fullTx bool) (interface{}, error)
 	return toBlock(block, fullTx), nil
 }
 
+// GetHeaderByNumber returns the requested canonical block header.
+// * When blockNr is -1 the chain head is returned.
+// * When blockNr is -2 the pending chain head is returned.
+func (e *Eth) GetHeaderByNumber(number BlockNumber) (interface{}, error) {
+	num, err := GetNumericBlockNumber(number, e.store)
+	if err != nil {
+		return nil, err
+	}
+
+	block, ok := e.store.GetBlockByNumber(num, true)
+	if !ok {
+		return nil, nil
+	}
+
+	if err := e.filterExtra(block); err != nil {
+		return nil, err
+	}
+
+	return toHeader(block.Header), nil
+}
+
+// GetHeaderByHash returns the requested header by hash.
+func (e *Eth) GetHeaderByHash(hash types.Hash) (interface{}, error) {
+	block, ok := e.store.GetBlockByHash(hash, true)
```

Comment:
Expose this function to the `ethBlockchainStore` interface and use it instead. We do not need the block body here.

```go
func (b *Blockchain) GetHeaderByHash(hash types.Hash) (*types.Header, bool)
```